### PR TITLE
Service updates and patches

### PR DESCRIPTION
**Pull request by ralph-hawkins**
Adds updates to the service.

Home page shows one latest update:

localhost_3000_start-page(screenshot)

Updates page shows all:

localhost_3000_service-updates(screenshot)

Updates to signed-in homepage:

localhost_3000_home(screenshot)

Note — content for updates is placeholder.

### CODE REVIEW

File: app/assets/sass/application.scss [hunk 1]
```diff
@@ -221,3 +221,7 @@ span.summary-validation-symbol {
 .app-hidden-form {
   display: inline;
 }
+
+ul {
```
Comment (edwardhorsford) [inline app/assets/sass/application.scss]: I think this should be scoped to a `app-markdown` class or something.

File: app/data/service-patches-and-updates.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+      "title": "Bursaries",
+      "content": "You can record which trainees are applying for a bursary."
+    },
+    {
+      "date": "2021-08-10",
+      "title": "Teaching apprenticeship (postgrad)",
+      "content": "You can now register trainee teachers to Early years (postgrad) training courses. **If** you need help contact"
+    }
+  ]
+}
```
Comment (edwardhorsford) [inline app/data/service-patches-and-updates.json]: Not sure this file is wanted any more.

File: app/data/service-patches-and-updates.yaml [hunk 3]
```diff
@@ -0,0 +1,35 @@
+updates:
```
Comment (edwardhorsford) [inline app/data/service-patches-and-updates.yaml]: Could the filename be just `service-updates.yaml`? Patches feels rather technical

File: app/data/session-data-defaults.js [hunk 4]
```diff
@@ -162,6 +168,7 @@ module.exports = {
   publishRoutes,
   nonPublishRoutes,
   ukComparableDegrees,
+  updatesAndPatches,
```
Comment (edwardhorsford) [inline app/data/session-data-defaults.js]: Would probably call this `serviceUpdates` or `serviceUpdatesAndNews` or something.

File: app/filters/strings.js [hunk 5]
```diff
@@ -71,6 +72,27 @@ filters.currency = input => {
...
+    // },
+    // langPrefix: 'hljs language-', // highlight.js css expects a top-level 'hljs' class.
+    // pedantic: false,
+    // gfm: true,
+    // breaks: false,
+    // sanitize: false,
+    // smartLists: true,
+    // smartypants: false,
+    // xhtml: false
+  })
```
Comment (edwardhorsford) [inline app/filters/strings.js]: Could probably delete all these comments?

File: app/views/service-updates.html [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+<div class="govuk-grid-row">
+  <div class="govuk-grid-column-full">
+    <div class="govuk-grid-row">
+      <div class="govuk-grid-column-two-thirds-from-desktop">
+        <h1 class="govuk-heading-xl">{{pageHeading}}</h1>
+        <h2 class="govuk-heading-m">Contents</h2>
+        <ul class="govuk-list govuk-list--dash govuk-!-margin-bottom-8 govuk-!-margin-left-3">
+          {% for update in data.updatesAndPatches.updates %}
+            <li><a class="govuk-link" href="#{{ update.title | escape }}">{{ update.title }}</a></li>
```
Comment (edwardhorsford) [inline app/views/service-updates.html]: What's this `escape` filter? Don't we want the title camel-cased for the url?

File: app/data/service-patches-and-updates.yaml (hunk 3 above)
Comment (edwardhorsford) [inline app/data/service-patches-and-updates.yaml]:
```suggestion
items:
```

File: app/views/service-updates.html [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+          {% endfor %}
+        </ul>
+
+
+
+        {% for update in data.updatesAndPatches.updates %}
+          <hr class="govuk-section-break govuk-section-break--l govuk-section-break--visible">
+          <section id="section-about-register">
```
Comment (edwardhorsford) [inline app/views/service-updates.html]: This id should presumably be the title, camel-cased (there's already a filter to do this).

File: app/views/service-updates.html [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+            <h2 class="govuk-heading-m govuk-!-margin-bottom-2" id="{{ update.title | escape }}">
```
Comment (edwardhorsford) [inline app/views/service-updates.html]: I can see there's an ID on this h2 - wonder if this should be on the parent section instead?

**Review by edwardhorsford — CHANGES_REQUESTED** [review]

**Review by gregknight1 — COMMENTED** [review]
Looks really good @ralph-hawkins. I take it there's no issue in having a link on the sign-in page. What's springing to mind is how the full list of news/updates is accessed when a user is signed in?

Wondering how we'd approach an update with a large amount of content or multiple updates at the same time.

Also thinking that the contents could end up being massive, as could the actual updates. Think we need some rules around the maintenance of this.

**Review by edwardhorsford — APPROVED** [review]

File: app/views/service-updates.html [hunk 9]
```diff
@@ -0,0 +1,38 @@
+
+{% extends "_templates/_page.html" %}
+{% set backLink = '/start-page' %}
```
Comment (edwardhorsford) [inline app/views/service-updates.html:3]: This should probably be js back - `/start-page` isn't the only source for this page.

**Pull request merged**